src/init.cpp

```diff
@@ -1003,10 +1012,12 @@ bool AppInitParameterInteraction()
         return InitError(_("Cannot set -peerblockfilters without -blockfilterindex."));
     }
 
-    // if using block pruning, then disallow txindex
+    // if using block pruning, then disallow txindex, coinstatsindex and blockfilterindex
     if (gArgs.GetArg("-prune", 0)) {
         if (gArgs.GetBoolArg("-txindex", DEFAULT_TXINDEX))
             return InitError(_("Prune mode is incompatible with -txindex."));
+        if (gArgs.GetBoolArg("-coinstatsindex", DEFAULT_COINSTATSINDEX))
+            return InitError(_("Prune mode is incompatible with -coinstatsindex."));
```

Comment:
Why?

Reply:
It syncs over the whole blockchain, processing every block and saving state on each. That allows users to query the stats for every block height (a nice-to-have feature). It's the standard way `BaseIndex` currently works, although changing that would probably only require minimal effort. Potential users who I have talked to didn't really care much about this. But I will look into it if this again since progress is slow anyway currently.

Reply:
As a potential user, it would be nice if I can at least turn on pruning _after_ the index is generated, but I think that's for a separate PR.